app/components/form/fields/TlsCertsField.tsx

```diff
@@ -147,3 +176,88 @@ const AddCertModal = ({ onDismiss, onSubmit, allNames }: AddCertModalProps) => {
     </Modal>
   )
 }
+
+const validateCertificate = async (file: File) => {
+  return parseCertificate(await file.text())
+}
+
+function parseCertificate(certPem: string) {
+  try {
+    const cert = new X509Certificate(certPem)
+    const nameItems = cert.getExtension(SubjectAlternativeNameExtension)?.names.items || []
+    return {
+      commonName: cert.subjectName.getField('CN') || [],
+      subjectAltNames: nameItems.map((item) => item.value) || [],
+    }
+  } catch {
+    return null
+  }
+}
+
+function matchesDomain(pattern: string, domain: string): boolean {
+  const patternParts = pattern.split('.')
+  const domainParts = domain.split('.')
+
+  if (patternParts.length !== domainParts.length) return false
+
+  return patternParts.every(
+    (part, i) => part === '*' || part.toLowerCase() === domainParts[i].toLowerCase()
```

Comment:
Would it just be the first part that accepts a wildcard?

Reply:
I presume it would be the left-most wildcard subdomain is the only allowable?

Would need adjusting, and some unit tests also.

Reply:
Correct only the left most subdomain can be a wildcard. For example `*.r3.oxide-preview.com`  does not match `silo.sys.r3.oxide-preview.com`